model_building/model_building.py

```diff
@@ -184,7 +192,11 @@ def process(self, campaign_configuration, regression_inputs, processes_number):
 
             if 'save_training_regressors' in campaign_configuration['General'] and campaign_configuration['General']['save_training_regressors']:
                 # Build the regressor trained on the train set only
-                training_regressor = regressor.Regressor(campaign_configuration, best_conf.get_regressor(), best_conf.get_x_columns(), regression_inputs.scalers)
+                if hasattr(best_conf, '_wrapped_experiment_configuration'):
+                    training_regressor = regressor.Regressor(campaign_configuration, best_conf.get_regressor(), best_conf.get_x_columns(), regression_inputs.scalers, best_conf._wrapped_experiment_configuration._hyperparameters)
+                else:
+                    training_regressor = regressor.Regressor(campaign_configuration, best_conf.get_regressor(), best_conf.get_x_columns(), regression_inputs.scalers, best_conf._hyperparameters)
+                
```

Comment:
For the sake of clarity, I'd rather do something like:
```py
hypers = best_conf._wrapped_experiment_configuration._hyperparameters if hasattr(best_conf, '_wrapped_experiment_configuration') else best_conf._hyperparameters
training_regressor = regressor.Regressor(..., hypers)
```

Reply:
Refactored, here and some lines after